Do not require ninja. We're not using it.
Reverts part of https://github.com/mongodb/mongo/commit/8d21d154677e2202e2a7ddd7f99d2a0ffedf4d53
--- etc/pip/components/compile.req.orig	2022-11-10 11:38:11.000000000 -0600
+++ etc/pip/components/compile.req	2023-01-11 20:05:47.000000000 -0600
@@ -3,4 +3,3 @@
 regex <= 2021.11.10
 requirements_parser <= 0.3.1
 setuptools
-ninja>=1.10.0
